client/src/components/related_products/RelatedProductsMain.jsx

```diff
@@ -19,32 +18,29 @@ function RelatedProductsMain() {
       })
   }
 
-  const getRelated = () => {
+  useEffect(() => {
     axios.get(`/products/${productID}/related`)
       .then((response) => {
-        var productsPromises = response.data.map((product) => {
-          return getProduct(product)
-        })
-        Promise.all(productsPromises)
-          .then((values) => {
-            setRelatedProducts(values);
-          })
-
-        //setRelatedProducts(response.data);
+        const uniqueIDs = _.uniq(response.data);
+        const filteredIDs = uniqueIDs.filter((id) => id !== productID);
+        const productsPromises = filteredIDs.map((product) => getProduct(product));
+        return productsPromises;
       })
-      .catch(() => {
-        console.log('Unable to retrieve related products');
+      .catch(() => console.log('Unable to retrieve related ids'))
```

Comment:
If an error is caught by the first .catch block, the next then blocks will not execute. However, the following catch will still execute. This might not be the desired behavior. Not sure if this promise chain is designed to be like this.